### PR TITLE
Quest Update 4-18-23 | Added Creative, Thermal, and Mystical Ag Quest Rework

### LINKED ISSUE #997: Make Chemical Oxidizer quest to accept Sulfur Dust from Thermal Series
It's easier to get with Mystical Agriculture.

image

### LINKED ISSUE #1210: Twilight Forest questline: Fancy Noms quest missing Torchberries
Fancy Noms quest missing Torchberries as a food which needed to complete "We Dine At Eternal Sundown" advancement.

image

image

### PR DESCRIPTION
**Pull request by alfredggttv**
- Added Creative Questline
- Ported Thermal Chapter, reworked rewards and a few quests
- Added Alloys and Alloy Tools to Allthemodium, reworked some rewards
- Complete rework of Mystical Ag Chapter, added custom random rewards

**Fixes**
- Update Several Twilight Forest Quest descriptions: Fix for #1105 and added Torchberries to fix #1210
- Update AE2 Crystal Growth Accelerator quest: Fix for #1080
- Fix Requirements for ElementalCraft Rune Quests - Fix for #1015
- Fix Mekanism quest to accept "all forge sulfur dusts" - Fixes #997
- Changed Questline Titles to prep for future update
- Fix quest in RS for the Advanced Storage Housing to allow players to unlock the quest from either fluid or item storage parts, rather than needing both.
- Adjusted Random Rewards for balance
- Minor changes and updates to Tips and Tricks

### CODE REVIEW

File: config/ftbquests/quests/chapters/creative.snbt [hunk 1]
```diff
@@ -0,0 +1,817 @@
...
+				}
+			]
+		}
+		{
+			title: "Set For Life"
+			x: 0.0d
+			y: 12.0d
+			shape: "hexagon"
+			subtitle: "Enough to make everything in the pack?"
+			description: ["If you make 9 Stars, or enough to make an ATM Star Block, you'll be able to create a &dStarry Bee&r. This bee gives you &6ATM Star Shards&r from their combs."]
```
Comment (James103) [inline config/ftbquests/quests/chapters/creative.snbt:584]: Last time I checked, the Starry Bee costs 1 ATM Star Block to create and requires another ATM Star Block to use as its flower, a total of 2 ATM Star Blocks (18 ATM Stars), not 1 block (9 stars).

File: config/ftbquests/quests/chapters/thermal_expansion.snbt [hunk 2]
```diff
@@ -0,0 +1,1775 @@
...
+				type: "random"
+				exclude_from_claim_all: true
+				table_id: 4084485630345500261L
+			}]
+		}
+		{
+			x: 8.0d
+			y: 2.0d
+			shape: "hexagon"
+			subtitle: "Increases Generation Rate at Cost of Efficiency"
```
Comment (James103) [inline config/ftbquests/quests/chapters/thermal_expansion.snbt:1271]:
```suggestion
			subtitle: "Increases Generation Rate at the Cost of Efficiency"
```

File: config/ftbquests/quests/chapters/creative.snbt (hunk 1 above)
Comment (alfredggttv) [reply config/ftbquests/quests/chapters/creative.snbt]: Forgot to change this. Thanks for pointing this out!

**Comment by JuliusGudni** [comment]
Not sure if intended but it seems to me that the PR is missing the actual refactored mystical agriculture chapter mentioned in the PR description

**Comment by alfredggttv** [comment]
> Not sure if intended but it seems to me that the PR is missing the actual refactored mystical agriculture chapter mentioned in the PR description

Got this in! Had a mini heart attack thinking it got deleted. Thanks for pointing it out!

**Comment by JuliusGudni** [comment]
Also noticed that the coal seed quest has a broken random reward :)

**Review by Mitchell5200 — APPROVED** [review]

**Pull request merged**